lib/ansible/module_utils/network/nxos/config/l2_interfaces/l2_interfaces.py

```diff
@@ -255,7 +255,10 @@ def add_commands(self, d):
         if 'vlan' in d:
             commands.append(cmd + 'access vlan ' + str(d['vlan']))
         if 'allowed_vlans' in d:
-            commands.append(cmd + 'trunk allowed vlan ' + str(d['allowed_vlans']))
+            if vlan_exists:
```

Comment:
@GomathiselviS code changes LGTM, can u add on 1 test covering the use case.

Reply:
Added testcase